api/v1beta1/verticarestorepointsquery_types.go

```diff
@@ -23,21 +23,43 @@ import (
 
 // VerticaRestorePointsQuerySpec defines the desired state of VerticaRestorePointsQuery
 type VerticaRestorePointsQuerySpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
-
 	// +kubebuilder:validation:Required
 	// +operator-sdk:csv:customresourcedefinitions:type=spec
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:com.tectonic.ui:text"
 	// The name of the VerticaDB CR that this VerticaRestorePointsQuery is defined for.  The
 	// VerticaDB object must exist in the same namespace as this object.
 	VerticaDBName string `json:"verticaDBName"`
 
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:com.tectonic.ui:advanced"
+	// Optional parameter that will limit the query to only restore points satisfying provided filter options
+	FilterOptions *VerticaRestorePointQueryFilterOptions `json:"filterOptions,omitempty"`
```

Comment:
I'm not sure, the design doc only mentions these 3 filters

Reply:
That's a good idea. I think you mean using `ShowRestorePointFilterOption`? But I think we'll need the markers added to it. We were able to avoid adding them for the status because there is no OpenShift UI to add the status. But we'll need them for the spec.

Reply:
Yes I mean that one.

Reply:
Looking at @jizhuoyu latest changes, I'm starting to think the current approach is fine. We duplicate the FilterOptions struct. That isn't good, but we would need to include operator-sdk markers in the vclusterops package. And it is a safe guard against someone inadvertently renaming/removing fields in the vclusterops package without understanding the downstream impact. Lets discuss in the stand-up today.